samples/examples/FSharpExamples/Matrix.fs

```diff
@@ -28,38 +28,53 @@ open Prajna.Api.FSharp
 
 open Prajna.Examples.Common
 
-type MatrixSample() =
+module Matrix =
 
     let (.*) (xs: float[]) (ys: float[]) = 
-        // simulate doing some more work, 
-        // e.g.: run the remaining layers of a neural net
-        Threading.Thread.Sleep(3) 
         Array.fold2 (fun sum x y -> sum + x * y) 0.0 xs ys 
 
     let printTiming (msg: string) (x: Lazy<'b>) : 'b =
         let sw = Stopwatch.StartNew()
+        printf "%s: " msg
         let ret = x.Value
         sw.Stop()
-        printfn "%s: %A" msg sw.Elapsed
+        printfn "%A" sw.Elapsed
         ret
 
-    let run (cluster: Cluster) =
+    let random = new Random(0)
+
+    let randomVector n =  Array.init n (fun _ -> random.NextDouble())
+    
+    let randomMatrix m n =  Array.init m (fun _ -> randomVector n)
+
+    /// Matrix-Vector multiplication for illustrational purposes.
+    /// The overhead of distributing makes it faster to run this locally.
+    /// See Matrix-Matrix for a case where the computation is actually faster distributed.
+    let matrixVector (cluster: Cluster) =
 
         let numRows = 200
         let vectorSize = 100
 
-        let r = new Random(0)
-        let randomMatrix =  Array.init numRows (fun _ -> Array.init vectorSize (fun _ -> r.NextDouble()))
-        let vector = Array.init vectorSize (fun _ -> r.NextDouble())
+        let matrix =  randomMatrix numRows vectorSize
+        let vector = randomVector vectorSize
 
         let singleCoreResult = 
             printTiming "single core" <| lazy 
-                randomMatrix |> Array.map (fun row -> row .* vector)
-        
+                matrix |> Array.map (fun row -> row .* vector)
+
+        let numPartitions = Environment.ProcessorCount
```

Comment:
Ok. What about something simple like cluster.NumNodes * 30? Alternatively, we could run a precomputation on all nodes to extract the machine name and number of processors on each node, then add the nodes per unique machine. But I don't think our APIs even guarantee distribution, so we would have to either add that guarantee to an API (maybe DSet.distribute?) or go lower level to extract the ideal counts. Any preference? Other options?